CARDS-2734: Allow specifying more than one node property filter in the pagination servlet

veronikaslc opened the pull request:
Specs: [CARDS-2734](https://phenotips.atlassian.net/browse/CARDS-2734)

- Remaned filed params names to be plural
- Added `fieldcomparators` of `=` to all frontend field param query usage
- Refactored backend to support multivalued field names, values, comparators if equally provided
- Altered `parseFiltersFromRequest()` to check for null case of no filters which led to refactoring on line 503 to avoid `CyclomaticComplexity` checkstyle error

[CARDS-2734]: https://phenotips.atlassian.net/browse/CARDS-2734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

veronikaslc committed:
CARDS-2734: Allow specifying more than one node property filter in the pagination servlet

sdumitriu commented on modules/data-entry/src/main/java/io/uhndata/cards/PaginationServlet.java:
```diff
@@ -429,19 +437,32 @@ protected String createQuery(final SlingHttpServletRequest request, Session sess
+
+        final Map<String, String[]> fieldParameters = new HashMap<>();
+        // if field names and values are provided
+        if (names != null && values != null && comparators != null) {
```
Unexpected behavior: when one of new parameters is completely missing, all of them are completely ignored. The rule should be: if any of them is present, check if they are all present and have the same length. Use `||` to join the conditions above, then check if any is null below.

veronikaslc committed:
CARDS-2734: Allow specifying more than one node property filter in the pagination servlet
if any of them is present, check if they are all present and have the same length
